Cargo.lock

```diff
@@ -24,6 +24,21 @@ dependencies = [
  "nodrop 0.1.13 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
+[[package]]
+name = "asdl-rs"
+version = "0.1.0"
+source = "git+https://github.com/pasa/asdl-rs.git#5c4200989396dad596472c795e6c728cf6f1537c"
+dependencies = [
+ "clap 2.33.0 (registry+https://github.com/rust-lang/crates.io-index)",
+ "failure 0.1.5 (registry+https://github.com/rust-lang/crates.io-index)",
+ "heck 0.3.1 (registry+https://github.com/rust-lang/crates.io-index)",
+ "nom 5.0.0 (registry+https://github.com/rust-lang/crates.io-index)",
+ "rustc-hash 1.0.1 (registry+https://github.com/rust-lang/crates.io-index)",
+ "serde 1.0.98 (registry+https://github.com/rust-lang/crates.io-index)",
+ "serde_json 1.0.40 (registry+https://github.com/rust-lang/crates.io-index)",
+ "tera 0.11.20 (registry+https://github.com/rust-lang/crates.io-index)",
+]
```

Comment:
I haven't looked at the asdl-rs source code yet, but `failure`, `tera`, `clap` and `serde_json` deps looks like they ideally shouldn't belong to the ASDL *library*.

Reply:
Yeah. There are used for CLI interface. I will check how to make separate dependencies for lib